subprojects/ide-native/src/integTest/groovy/org/gradle/ide/visualstudio/VisualStudioSoftwareModelIncrementalIntegrationTest.groovy

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.visualstudio
+
+import org.gradle.ide.visualstudio.fixtures.AbstractVisualStudioIntegrationSpec
+import org.gradle.integtests.fixtures.SourceFile
+import org.gradle.nativeplatform.fixtures.app.CppHelloWorldApp
+
+import static org.apache.commons.io.FileUtils.copyFile
+
+class VisualStudioSoftwareModelIncrementalIntegrationTest extends AbstractVisualStudioIntegrationSpec {
```

Comment:
I was trying to suppress my urge to test _everything_ and only have separate tests for the things that are known to have subtle differences between the models.  I was explicitly trying not to duplicate the things that should reasonably work the same regardless of the model.  I'm not sure how well a shared test spec would work because of the subtle difference in naming and (not so subtle) difference in configuration between the models, anyways.  I'm prepared to be convinced otherwise, though.

Reply:
I'm with Gary on this -- we should try to cover things that are different and only combine tests for the software model and current model plugins where it's easy to do so. A little apparent duplication will make it easier later if/when we move the software model tests into their own bucket.

Reply:
I trust your judgment on this. We also don't want to drag those test for too long anyway, so it's all good.